modules/miner/cpuminer.go

```diff
@@ -2,11 +2,18 @@ package miner
 
 import (
 	"time"
+
+	"github.com/NebulousLabs/Sia/build"
 )
 
 // threadedMine starts a gothread that does CPU mining. threadedMine is the
 // only function that should be setting the mining flag to true.
 func (m *Miner) threadedMine() {
+	if err := m.tg.Add(); err != nil {
+		return
+	}
+	defer m.tg.Done()
+
```

Comment:
I'm re-writing some previous comments because they had an incorrect understanding of how `tg` works.

Definitely okay to call `tg.Add()` without holding a miner lock.